README.md

```diff
@@ -1,43 +1,73 @@
+**NOTE**: All of the APIs are defined under the API group `objectstorage.k8s.io`.
 
+For more information about COSI, visit our [documentation](https://github.com/kubernetes-sigs/container-object-storage-interface-api/tree/master/docs/index.md).
 
-# Container Object Storage Interface (COSI) API
+## Developer Guide
 
-API definitions for Container Object Storage.
+All API definitions are in [`apis/objectstorage.k8s.io/`](./apis/objectstorage.k8s.io/). All API changes **_MUST_** satisfy the following requirements:
 
-## Community, discussion, contribution, and support
+ - Must be backwards compatible
+ - Must be in-sync with the API definitions in [sigs.k8s.io/container-object-storage-interface-spec](https://github.com/kubernetes-sigs/container-object-storage-interface-spec)
+ 
+### Build and Test
+
+1. Test and Build the project
 
-If you are new to a SIG Storage project, check out this [contributing guide](https://github.com/kubernetes/community/blob/master/sig-storage/CONTRIBUTING.md) 
-and the [community page](https://github.com/kubernetes/community/tree/master/sig-storage). 
+```
+make all
+```
 
-If you are new to the SIG Storage COSI project, check out the [spec](https://github.com/kubernetes-sigs/container-object-storage-interface-spec/blob/master/spec.md), [KEP](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/1979-object-storage-support), and [project board](https://github.com/orgs/kubernetes-sigs/projects/).
+2. Generate CRDs
 
-Learn how to engage with the Kubernetes community on the [community page](http://kubernetes.io/community/).
+```
+make codegen
+```
+
+## Adding new fields to protocols
+
+1. Create a new issue raising a RFC for the changes following this format:
+
+Title: [RFC] Changes to protocol xyz 
+> **Info**:
+> 1. Protocol: 
+> 2. Fields Added:
+> 3. Why is this change neccessary?
+>    ...(describe why here)...
+> 4. Which other COSI projects are affected by this change?
+> 5. Upgrade plan 
+>    (ignore if it doesn't apply)
+
+## References
+
+ - [Documentation](docs/index.md)
+ - [Deployment Guide](docs/deployment-guide.md) 
+ - [Weekly Meetings](docs/meetings.md)
+ - [Roadmap](https://github.com/orgs/kubernetes-sigs/projects/8)
+
+## Community, discussion, contribution, and support
 
 You can reach the maintainers of this project at:
 
-- [Slack](https://kubernetes.slack.com/messages/sig-storage-cosi)
-- [Mailing List](https://groups.google.com/g/container-object-storage-interface-wg?pli=1)
+ - [#sig-storage-cosi](https://kubernetes.slack.com/messages/sig-storage-cosi) slack channel 
+ - [container-object-storage-interface](https://groups.google.com/g/container-object-storage-interface-wg?pli=1) mailing list
 
 ### Code of conduct
 
-Participation in the Kubernetes community is governed by the [Kubernetes Code of Conduct](code-of-conduct.md).
\ No newline at end of file
+Participation in the Kubernetes community is governed by the [Kubernetes Code of Conduct](code-of-conduct.md).
```

Comment:
This link is one level more direct: https://github.com/kubernetes/community/blob/master/code-of-conduct.md

Reply:
but maybe you prefer the link to be inside the repo here?